server/core/static/external/cred_update.js

```diff
@@ -0,0 +1,106 @@
+// Makes the password form interactive (e.g. shows when passwords don't match)
+function setupInteractivePwdFormListeners() {
+    const new_pwd = document.getElementById("new-password");
+    const new_pwd_check = document.getElementById("new-password-check");
+    const pwd_submit = document.getElementById("password-submit");
+
+    function markPwdCheckValid() {
+        new_pwd_check.classList.remove("is-invalid");
+        new_pwd_check.classList.add("is-valid");
+        pwd_submit.disabled = false;
+    }
+
+    function markPwdCheckInvalid() {
+        new_pwd_check.classList.add("is-invalid");
+        new_pwd_check.classList.remove("is-valid");
+        pwd_submit.disabled = true;
+    }
+
+    new_pwd.addEventListener("input", (_) => {
+        // Don't mark invalid if user didn't fill in the confirmation box yet
+        // Also my password manager (keepassxc with autocomplete)
+        //   likes to fire off input events when both inputs were empty.
+        if (new_pwd_check.value !== "") {
+            if (new_pwd.value === new_pwd_check.value) {
+                markPwdCheckValid();
+            } else {
+                markPwdCheckInvalid();
+            }
+        }
+        new_pwd.classList.remove("is-invalid");
+    });
+
+    new_pwd_check.addEventListener("input", (_) => {
+        // No point in updating the status if confirmation box is empty
+        if (new_pwd_check.value === "") return;
+        if (new_pwd_check.value === new_pwd.value) {
+            markPwdCheckValid();
+        } else {
+            markPwdCheckInvalid();
+        }
+    });
+}
+
+function stillSwapFailureResponse(event) {
+    if (event.detail.xhr.status === 422) {
+        console.log("Still swapping failure response")
+        event.detail.shouldSwap = true;
+        event.detail.isError = false;
+    }
+}
+
+function startPasskeyEnrollment() {
+    let data_elem = document.getElementById('data');
+    let credentialRequestOptions = JSON.parse(data_elem.textContent);
```

Comment:
If it's never reassigned, it should be a const

```suggestion
    let credentialRequestOptions = JSON.parse(data_elem.textContent);
```